core/templates/components/skill-selector/skill-selector.component.spec.ts

```diff
@@ -0,0 +1,382 @@
+    component.currCategorizedSkills = {
+      topic1: {
+        uncategorized: [
+          ShortSkillSummary.create('skill1', 'Skill 1 description.')
+        ],
+        subtopic1: [
+          ShortSkillSummary.create('skill2', 'Skill 2 description.')
+        ]
+      },
+      topic2: {
+        uncategorized: []
+      }
+    };
+
+    expect(component.checkTopicIsNotEmpty('topic1')).toBe(true);
+    expect(component.checkTopicIsNotEmpty('topic2')).toBe(false);
+  });
+
+  it('should set selected skill Id when user clicks on radio button', () => {
+    component.selectedSkill = 'skill1';
+    spyOn(component.selectedSkillIdChange, 'emit');
+
+    component.setSelectedSkillId();
+
+    expect(component.selectedSkillIdChange.emit).toHaveBeenCalledWith('skill1');
+  });
+
+  it('should display subtopics from all topics in the subtopic filter if' +
+    ' no topic is checked', () => {
+    component.categorizedSkills = {
+      topic1: {
+        uncategorized: [
+          ShortSkillSummary.create('skill1', 'Skill 1 description.')
+        ],
+        subtopic1: [
+          ShortSkillSummary.create('skill2', 'Skill 2 description.')
+        ]
+      },
+      topic2: {
+        uncategorized: [
+          ShortSkillSummary.create('skill4', 'Skill 4 description.')
+        ]
+      }
+    };
+
+    component.ngOnInit();
+
+    component.subTopicFilterDict = {};
+
+    component.updateSkillsListOnTopicFilterChange();
+
+    // All subtopics from all topics is included in the subtopic filter dict.
+    expect((component.subTopicFilterDict)).toEqual({
+      topic1: [
+        {
+          subTopicName: 'uncategorized',
+          checked: false
+        },
+        {
+          subTopicName: 'subtopic1',
+          checked: false
+        }
+      ],
+      topic2: [
+        {
+          subTopicName: 'uncategorized',
+          checked: false
+        }
+      ]
+    });
+  });
+
+  it('should update skill list when user filters skills by only topics', () => {
+    component.categorizedSkills = {
+      topic1: {
+        uncategorized: [
+          ShortSkillSummary.create('skill1', 'Skill 1 description.')
+        ],
+        subtopic1: [
+          ShortSkillSummary.create('skill2', 'Skill 2 description.')
+        ],
+        subtopic2: [
+          ShortSkillSummary.create('skill3', 'Skill 3 description.')
+        ]
+      },
+      topic2: {
+        uncategorized: [
+          ShortSkillSummary.create('skill4', 'Skill 4 description.')
+        ],
+        subtopic3: [
+          ShortSkillSummary.create('skill5', 'Skill 5 description.')
+        ],
+        subtopic4: [
+          ShortSkillSummary.create('skill6', 'Skill 6 description.')
+        ]
+      }
+    };
+    component.ngOnInit();
+
+    // User checks the 'topic1' radio button.
+    component.topicFilterList = [
+      {
+        topicName: 'topic1',
+        checked: true
+      },
+      {
+        topicName: 'topic2',
+        checked: false
+      }
+    ];
+
```

Comment:
Can we add a pre-check here? Same for the other tests. It would be better to test the state change right?

Reply:
Added